opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -344,7 +344,8 @@ namespace Opm
                     eclIO_->overwriteInitialOIP(simProps);
                 }
                 // ... insert "extra" data (KR, VISC, ...)
-                eclIO_->writeTimeStep(timer.reportStepNum(),
+                const int reportStepForOutput = substep ? timer.reportStepNum() + 1 : timer.reportStepNum();
```

Comment:
Personally I would prefer to have the special casing here: https://github.com/OPM/opm-simulators/blob/160314424e75661cb7255722d91bb1746d3bf016/opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp#L328 - then the hardcoding of `substep = true;` and `report_step + 1` would be located at the same place in the code. But it will be slightly ugly however it is done.

Reply:
At that point we pass the timer object rather than the index, so it would be more involved (and potentially obfuscated) to do it there.

Reply:
> At that point we pass the timer object

I realized that ....